exporter/datasetexporter/examples/e2e/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM golang:1.19
```

Comment:
Why are we doing all this and not grabbing the latest contrib image?

Reply:
Because this exporter is not part of the official contrib image. So if you want to use `datasetexporter`, you have to build you own binary.

So for building collector which contains this code you you have to write your own `otelcol-builder.yaml` and include something like this as replacement:
```
replaces:
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datasetexporter v0.77.0 => github.com/scalyr/opentelemetry-collector-contrib/exporter/datasetexporter issue-20660-datasetexporter-logs-traces-initial-code
```

Purpose of this is to demonstrate how to use it.

Reply:
I would suggest deferring this to a future PR once the dataset exporter is added to contrib, to avoid having to do this. It should not take too long for this to happen, and it will be much simpler for us to maintain